cli/wordpress-server-child.ts

```diff
@@ -0,0 +1,187 @@
+			},
+		];
+
+		const args: RunCLIArgs = {
+			command: 'server',
+			internalCookieStore: false,
+			login: false,
+			followSymlinks: true,
+			skipSqliteSetup: true,
+			port: config.port,
+			'mount-before-install': mounts,
+			'site-url': config.absoluteUrl || `http://localhost:${ config.port }`,
+			blueprint: config.blueprint || {},
+			wordpressInstallMode: 'download-and-install',
+		};
+
+		if ( hasWordPress ) {
+			args.wordpressInstallMode = 'install-from-existing-files-if-needed';
+		}
+
+		if ( config.phpVersion ) {
+			args.php = config.phpVersion as SupportedPHPVersion;
+		}
+
+		if ( config.wpVersion ) {
+			if ( isWordPressDevVersion( config.wpVersion ) ) {
+				args.wp = 'nightly';
+			} else {
+				args.wp = config.wpVersion;
+			}
+		}
+
+		args.blueprint.constants = { ...args.blueprint.constants, ...defaultConstants };
+		const server = await runCLI( args );
+
+		if ( ! server ) {
+			throw new Error( 'Failed to start server: runCLI returned void' );
+		}
+
+		if ( config.adminPassword ) {
+			await setAdminPassword( server, config.adminPassword );
+		}
+	} catch ( error ) {
+		server = null;
+		console.error( `[WordPress Server Child] Failed to start server:`, error );
+		throw error;
+	}
+}
+
+if ( process.send ) {
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	process.on( 'message', async ( packet: any ) => {
+		try {
+			const message: Message = packet.type === 'process:msg' && packet.data ? packet.data : packet;
+
+			const validationResult = MessageSchema.safeParse( message );
+			if ( ! validationResult.success ) {
+				throw new Error(
+					`Invalid message format: ${ validationResult.error.errors
+						.map( ( e ) => e.message )
+						.join( ', ' ) }`
+				);
+			}
+
+			const validMessage = validationResult.data;
+			let result: unknown;
+
+			switch ( validMessage.type ) {
+				case 'start-server':
+					if ( validMessage.data?.config ) {
+						result = await startServer( validMessage.data.config );
+					}
+					break;
+				default:
+					throw new Error( `Unknown message type: ${ validMessage.type }` );
+			}
+
+			if ( process.send && validMessage.id !== undefined ) {
+				const response = { id: validMessage.id, type: validMessage.type, result };
+				const responseValidation = MessageSchema.safeParse( response );
+				if ( responseValidation.success ) {
+					process.send( response );
+				}
+			}
+		} catch ( error ) {
+			const messageId =
+				packet.type === 'process:msg' && packet.data?.id !== undefined ? packet.data.id : packet.id;
+
+			if ( process.send && messageId !== undefined ) {
+				const errorResponse = {
+					id: messageId,
+					type: 'error',
+					error: error instanceof Error ? error.message : String( error ),
+					errorStack: error instanceof Error ? error.stack : undefined,
+				};
+				const errorValidation = MessageSchema.safeParse( errorResponse );
+				if ( errorValidation.success ) {
+					process.send( errorResponse );
+				}
+			}
```

Comment:
```suggestion
			console.log( '[WordPress Server Child] Error:', error );

			const minimalMessageSchema = z.object( { id: z.number() } );
			const message = minimalMessageSchema.safeParse( packet );

			if ( message.success ) {
				const errorResponse: ResponseMessageError = {
					id: message.data.id,
					topic: 'error',
					error: error instanceof Error ? error.message : String( error ),
					errorStack: error instanceof Error ? error.stack : undefined,
				};
				process.send!( errorResponse );
			}
```
I suggest doing something like this:
- Logging the error as standard seems sensible
- We might as well make use of zod
- See my comment in `cli/lib/types/wordpress-server.ts` for the `ResponseMessageError` definition.